### PR TITLE
[CALCITE-6493] Add MySQL and other professional term restrictions

### PR DESCRIPTION
**Pull request by caicancai**

### CODE REVIEW

**Comment by caicancai** [comment]
I have tested it locally

**Comment by caicancai** [comment]
@julianhyde @zabetak The PR is ready. Can you help check it?

File: core/src/test/java/org/apache/calcite/test/LintTest.java [hunk 1]
```diff
@@ -393,6 +393,21 @@ private static void checkMessage(String subject, String body,
     if (subject2.matches("[a-z].*")) {
       consumer.accept("Message must start with upper-case letter");
     }
+    if (subject2.matches(".*\\bmysql\\b.*") || subject2.matches(".*\\bMysql\\b.*")) {
```
[the remark this answers is not in the document]
Comment (caicancai) [reply core/src/test/java/org/apache/calcite/test/LintTest.java:396]: get

File: core/src/test/java/org/apache/calcite/test/LintTest.java [hunk 2]
```diff
@@ -393,6 +395,27 @@ private static void checkMessage(String subject, String body,
     if (subject2.matches("[a-z].*")) {
       consumer.accept("Message must start with upper-case letter");
     }
+
+    // Check for keywords that should be capitalized
```
Comment (julianhyde) [inline core/src/test/java/org/apache/calcite/test/LintTest.java:399]: Consider moving keywordMap, and its population, outside the loop.

**Review by julianhyde — APPROVED** [review]
LGTM

File: core/src/test/java/org/apache/calcite/test/LintTest.java (hunk 2 above)
Comment (caicancai) [reply core/src/test/java/org/apache/calcite/test/LintTest.java:414]: done

**Review by suibianwanwank — APPROVED** [review]
LGTM

**Comment by caicancai** [comment]
no other comment, squashed.
@julianhyde @suibianwanwank Thank you for your review

**Comment by caicancai** [comment]
@julianhyde Is it good to merge? We can solve this problem in version 1.38.0

**Pull request merged**